src/gui/addnewtorrentdialog.cpp

```diff
@@ -362,14 +394,16 @@ void AddNewTorrentDialog::saveSavePathHistory() const
     QDir selectedSavePath(ui->savePath->selectedPath());
     // Get current history
     QStringList history = settings()->loadValue(KEY_SAVEPATHHISTORY).toStringList();
+    if (history.size() > savePathHistoryLength())
```

Comment:
It's much better now, but still not cropped during adv. setttings save, would be nice to extract this and call it in `AddNewTorrentDialog::setSavePathHistoryLength` too.

Reply:
Why do you want that?

Reply:
Is not more logic to you, to crop the history list as well, when the value in adv. settings changes? This is about personal preferences, we can let this as its now, np.

Reply:
I understand *what* do you want, thank you. I don't understand *why*. What will tolerate the additional code dependency?

Reply:
The reason why is from the view of a user, I have 20 items in save path and in combobox dropdown, I go to adv. settings, set that I want to have only 5 items in Save path, so I set Save path history length to 5, now I add new torrent, add new torrent dialog is opened, uncollapse combobox dopdown and I see there are still 20 items, that was the reason why I asked this change.
You can add optional trimHistory param to setter `AddNewTorrentDialog::setSavePathHistoryLength(int value, bool trimHistory = false)`